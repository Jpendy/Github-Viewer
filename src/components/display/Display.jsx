import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ name, followerCount, followingCount, githubLink }) => (
  <p>
    { `Name: ${name} - Follower Count: ${followerCount} - Following Count: ${followingCount} - Github Link: ${githubLink}` }
  </p>
);

Display.propTypes = {
  name: PropTypes.string.isRequired,
  followerCount: PropTypes.string.isRequired,
  followingCount: PropTypes.string.isRequired,
  githubLink: PropTypes.string.isRequired  
};

export default Display;
