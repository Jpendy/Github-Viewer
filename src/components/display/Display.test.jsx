import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(<Display name='sampleName' followerCount={1} followingCount={1} githubLink={'sample link'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
