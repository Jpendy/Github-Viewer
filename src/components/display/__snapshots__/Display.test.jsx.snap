// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<p>
  Name: sampleName - Follower Count: 1 - Following Count: 1 - Github Link: sample link
</p>
`;
